common/src/primitives/height.rs

```diff
@@ -1,8 +1,37 @@
 use parity_scale_codec_derive::{Decode, Encode};
+use std::fmt;
+use std::ops::Add;
 
 #[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Encode, Decode)]
 pub struct BlockHeight(u64);
 
+// Display should be defined for thiserr crate
+impl fmt::Display for BlockHeight {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl From<BlockHeight> for u64 {
+    fn from(block_height: BlockHeight) -> u64 {
+        block_height.inner()
+    }
+}
+
+impl From<u64> for BlockHeight {
+    fn from(w: u64) -> BlockHeight {
+        BlockHeight(w)
+    }
+}
+
+impl Add<u64> for BlockHeight {
+    type Output = Self;
+
+    fn add(self, other: u64) -> Self {
+        Self(self.0.checked_add(other).expect("overflow when adding BlockHeight to instant"))
```

Comment:
Are we OK with `expect` here? I'm not sure.

Reply:
Yes. This issue should never happen.

Reply:
The whole code was changed. I'm resolving this.